src/chains-config/cache/QueryFeeDetailsCache.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * When checking the cache there are 3 possible choices to return.
+ */
+enum QueryFee {
+	/**
+	 * queryFeeDetails is available
+	 */
+	available = 'available',
+	/**
+	 * queryFeeDetails is not available
+	 */
+	notAvailable = 'notAvailable',
+	/**
+	 * We dont know is queryFeeDetails is available
+	 */
+	unknown = 'unknown',
+}
+
+export class QueryFeeDetailsCache {
+	/**
+	 * Highest known runtime that doesn't have queryFeeDetails.
+	 */
+	private _versionWithoutCall;
+	/**
+	 * Lowest known runtime that has queryFeeDetails
+	 */
+	private _versionWithCall;
+
+	constructor(
+		versionWithoutCall: number | null,
+		versionWithCall: number | null
+	) {
+		this._versionWithoutCall = versionWithoutCall;
+		this._versionWithCall = versionWithCall;
+	}
+
+	public isQueryFeeDetailsAvail(specVersion: number) {
+		const { available, notAvailable, unknown } = QueryFee;
+
+		if (this._versionWithCall && specVersion >= this._versionWithCall) {
```

Comment:
Worth doing `this._versionWithCall !== null` too? Else spec versions of 0 may be mishandled.

Also on line 44?